### ISSUE TITLE
Problem when the password of the database ends with '.'

### ISSUE THREAD

**Issue by ljarevalo** [DOMjudge/domjudge#2502]
### Description of the problem
I want to inform you that there is a bug, I think, with the domserver installation. I have been searching for a week and I have not found any reference to it anywhere. In my case, I set the MySQL password that ends with the character '.' and the problem lies in the fact that when executing dj_setup_database, this character is treated as string concatenation, and consequently, the password doesn't work. I've solved the problem by changing the password.

### Your environment
DOMjudge version (e.g. 8.3.2)
Operating system / Linux distribution and version (e.g. Ubuntu 22.04 and Debian 12)
Webserver (e.g. Apache)

### Steps to reproduce
When the database passworkd ends with '.' and the command dj_setupdatabase is run: sudo bin/dj_setup_database -u root -r install, it doesn´t work since it is concatenated with more informatiion

sudo bin/dj_setup_database -u root -r install

Doctrine\DBAL\DBALException: "Malformed parameter "url".

**Comment by vmcj** [comment]
I can reproduce this in my local build. The problem is also there with a simple `noop` command so I suspect the problem has to be within our codebase (and not within `symfony_console`).

**Comment by vmcj** [comment]
Hi @ljarevalo,

I thought I could reproduce this locally but that was another issue. Can you provide more steps on how to reproduce? Did you use the tarball or docker?

I checked by starting a development container with our `docker-compose` and changing the passwords there to something ending with a `.`. That does work for me, also when I re-enter the container and use your specific invocation as you listed as steps to reproduce.

**Comment by ljarevalo** [comment]
HI @vmcj 

In docker, the installation works without any problems, however it only happened when the installations is made using a tarball installation. 
Best regards, Luis

**Comment by meisterT** [comment]
@eldering I assume this is fixed?

**Comment by eldering** [comment]
> [@eldering](https://github.com/eldering) I assume this is fixed?

Yes, see https://github.com/DOMjudge/domjudge/pull/3235 and https://github.com/DOMjudge/domjudge/commit/640945782efc3318119c171bef873d8ed87eeea1 for a set of characters that is now supported. Note that not all characters are supported yet, in particular not `:` as it is the separator in `etc/dbpasswords.secret` and IIRC tabs and other non-standard whitespace characters (space itself is fine).
